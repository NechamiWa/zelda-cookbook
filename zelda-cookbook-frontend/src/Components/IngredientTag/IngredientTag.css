.ingredient-name {
  font-weight: normal;
  padding: 10px;
  border-radius: 20px;
  margin: 0 20px;
  color: #1C3744;
  font-size: 1.5em; }
