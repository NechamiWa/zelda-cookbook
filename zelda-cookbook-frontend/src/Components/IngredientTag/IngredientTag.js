import React from 'react';
import PropTypes from 'prop-types';
import './IngredientTag.css';

const IngredientTag = ({ id, name, remove }) => {
  return (
    <div className="IngredientTag">
      <h1
        className="ingredient-name"
      >
        {name}
      </h1>
      <button className="button__clear-filter"
        onClick={remove}
        id={id}
      >
        X
      </button>
    </div>
  );
};

const { string, func } = PropTypes;

IngredientTag.propTypes = {
  id: string,
  name: string,
  remove: func,
};

export default IngredientTag;
