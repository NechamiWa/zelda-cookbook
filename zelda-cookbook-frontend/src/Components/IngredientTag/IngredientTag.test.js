/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import IngredientTag from './IngredientTag';

describe('IngredientTag', () => {
  let wrapper;
  let mockId = '1';
  let mockName = 'Apple'
  let mockRemove = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<IngredientTag 
      id={mockId}
      name={mockName}
      remove={mockRemove}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should call mockRemove when clicked', () => {
    wrapper.find('button').simulate('click')
    expect(mockRemove).toHaveBeenCalled()
  });
});