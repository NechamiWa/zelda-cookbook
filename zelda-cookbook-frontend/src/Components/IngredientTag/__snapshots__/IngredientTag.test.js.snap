// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientTag should match the snapshot 1`] = `
<div
  className="IngredientTag"
>
  <h1
    className="ingredient-name"
  >
    Apple
  </h1>
  <button
    className="button__clear-filter"
    id="1"
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
